jinja2/filters.py

```diff
@@ -705,7 +707,11 @@ def do_groupby(environment, value, attribute):
        attribute of another attribute.
     """
     expr = make_attrgetter(environment, attribute)
-    return sorted(map(_GroupTuple, groupby(sorted(value, key=expr), expr)))
+    if sort:
+        return sorted(map(_GroupTuple, groupby(sorted(value, key=expr), expr)))
```

Comment:
Yes, `sorted()` creates a new list, the items aren't copied however but passed by reference. But indeed the outer sort seems to be unneeded. Anyway, that logic doesn't change with this pull request. I will file a new pull request to get rid of the redundant sort.